cmd/docker/docker.go

```diff
@@ -309,6 +353,39 @@ func runDocker(dockerCli *command.DockerCli) error {
 	return cmd.Execute()
 }
 
+func initializeTracing() error {
+	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.TraceContext{}, propagation.Baggage{}))
+
+	exporter, err := autoexport.NewSpanExporter(context.Background())
+	if err != nil {
+		return fmt.Errorf("creating span exporter: %w", err)
+	}
+	sp := sdktrace.NewBatchSpanProcessor(exporter)
+
+	// create a default resource but specify the service name explicitly, or
+	// it will be `unknown_service:<name_of_exe>`
+	//
+	// N.B. a schemaless resource is used with the STABLE semconv value of
+	// `service.name` to avoid https://github.com/open-telemetry/opentelemetry-go/issues/2341
+	res, err := resource.Merge(
+		resource.Default(),
+		resource.NewSchemaless(
+			attribute.String("service.name", "docker-cli"),
+		),
+	)
+	if err != nil {
+		return fmt.Errorf("merging resource: %w", err)
+	}
```

Comment:
I realized this is actually redundant because we do `os.SetEnv("OTEL_SERVICE_NAME", cmd.Name())` if it's not set

You can just `res := resource.Default()`

Reply:
Ah, nice! Just pushed a commit (to be squash) with what I think you meant 👍